### PR TITLE
Purge using namespace std from libyul/optimiser

### PR DESCRIPTION
**Pull request by junaire**
Related issue: https://github.com/ethereum/solidity/issues/14403

### CODE REVIEW

**Comment by junaire** [comment]
CC @nikola-matic

**Comment by junaire** [comment]
Can you take a look at this? @nikola-matic

**Comment by nikola-matic** [comment]
> Can you take a look at this? @nikola-matic

Can you please resolve the conflicts, so I can merge it as soon it's reviewed (if everything's OK of course)?

**Comment by junaire** [comment]
> > Can you take a look at this? @nikola-matic
> 
> Can you please resolve the conflicts, so I can merge it as soon it's reviewed (if everything's OK of course)?

Sure I rebased it :)

File: libyul/optimiser/FunctionSpecializer.cpp [hunk 1]
```diff
@@ -86,34 +85,34 @@ FunctionDefinition FunctionSpecializer::specialize(
...
+					std::vector<TypedName>{newFunction.parameters[index]},
+					std::make_unique<Expression>(std::move(*argument))
 				}
 			);
 
 	newFunction.body.statements =
 		std::move(missingVariableDeclarations) + std::move(newFunction.body.statements);
 
-	// Only take those indices that cannot be specialized, i.e., whose value is `nullopt`.
+	// Only take those indices that cannot be specialized, i.e., whose value is `std::nullopt`.
```
Comment (nikola-matic) [inline libyul/optimiser/FunctionSpecializer.cpp]: This one is actually fine.

File: libyul/optimiser/Suite.cpp [hunk 2]
```diff
@@ -474,15 +474,15 @@ void OptimiserSuite::runSequence(string_view _stepAbbreviations, Block& _ast, bo
...
+	std::unique_ptr<Block> copy;
 	if (m_debug == Debug::PrintChanges)
-		copy = make_unique<Block>(std::get<Block>(ASTCopier{}(_ast)));
-	for (string const& step: _steps)
+		copy = std::make_unique<Block>(std::get<Block>(ASTCopier{}(_ast)));
+	for (std::string const& step: _steps)
 	{
 		if (m_debug == Debug::PrintStep)
-			cout << "Running " << step << endl;
+			std::cout << "Running " << step << "\n";
```
Comment (nikola-matic) [inline libyul/optimiser/Suite.cpp]:
```suggestion
			std::cout << "Running " << step << std::endl;
```

Same thing on lines 498, 501 and 502. With these `std::` PRs, please avoid making any other changes.

**Review by nikola-matic — COMMENTED** [review]
Aside from the `endl` -> `"\n"` change, which you should revert, everything else is fine, so we can merge this as soon as you fix it (also make sure to squash the commits into one).

Also, if you're interested in contributing further in this manner - the `test` directory (and its subdirectories) will also need to be adapted to use the `std::` namespace. I'm gonna go over the dirs to be changed, and add them to the main issue (https://github.com/ethereum/solidity/issues/14403). Thanks!

**Review by nikola-matic — APPROVED** [review]

File: libyul/optimiser/FunctionSpecializer.cpp (hunk 1 above)
Comment (junaire) [reply libyul/optimiser/FunctionSpecializer.cpp]: Done

File: libyul/optimiser/Suite.cpp (hunk 2 above)
Comment (junaire) [reply libyul/optimiser/Suite.cpp]: Done

**Pull request merged**